src/Sylius/Bundle/ApiBundle/Doctrine/QueryCollectionExtension/CountryCollectionExtension.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Doctrine\QueryCollectionExtension;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\ContextAwareQueryCollectionExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
+use Sylius\Bundle\ApiBundle\Serializer\ContextKeys;
+use Sylius\Component\Addressing\Model\CountryInterface;
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final class CountryCollectionExtension implements ContextAwareQueryCollectionExtensionInterface
+{
+    private UserContextInterface $userContext;
+
+    public function __construct(UserContextInterface $userContext)
+    {
+        $this->userContext = $userContext;
+    }
+
+    public function applyToCollection(
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        string $operationName = null,
+        array $context = []
+    ): void {
+        if (!is_a($resourceClass, CountryInterface::class, true)) {
+            return;
+        }
+
+        $user = $this->userContext->getUser();
+        if ($user instanceof AdminUserInterface && in_array('ROLE_API_ACCESS', $user->getRoles(), true)) {
+            return;
+        }
+
+        Assert::keyExists($context, ContextKeys::CHANNEL);
+        $channel = $context[ContextKeys::CHANNEL];
+
+        if ($channel !== null && $channel->getCountries()->count() > 0) {
```

Comment:
As we've introduced assert for channel existence, we can remove this check 
```suggestion
        if ($channel->getCountries()->count() > 0) {
```